From 239e3c5ca3bdb67fffde97e891919f2da616075f Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 19 Feb 2025 17:59:37 +0100
Subject: [PATCH 1541/1695] dt-bindings: mfd: define st,iocomp property for
 pwr-regu node

st,iocomp property allows to set a fixed compensation code. If the
property is not present in the pwr_regu node, the automatic
compensation is enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 08fc1057b58141c82d36b685d37597780b96bce1 dt-bindings: mfd: define st,iocomp property for pwr-regu node]
Change-Id: Ia02d056ba313f55b2dd1f2bd6efe71c23fdfa5b0
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438576
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../devicetree/bindings/mfd/st,stm32mp25-pwr.yaml        | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml b/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml
index 3af8401f3..8267c6932 100644
--- a/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml
+++ b/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml
@@ -43,6 +43,15 @@ patternProperties:
       vdd-supply:
         description: Input supply phandle(s) for vdd and vddio input
 
+      st,iocomp:
+        $ref: "/schemas/types.yaml#/definitions/uint32-array"
+        description:
+          The fixed compensation code in the following order <PMOS NMOS>.
+          If the property is not present in the node, the automatic compensation
+          is enabled.
+        minItems: 2
+        maxItems: 2
+
     patternProperties:
       ".*-supply$":
         description: Input supply phandle(s) for this node
-- 
2.47.2

